diff --git a/common.gypi b/common.gypi
index 8cf11845..b5f0051d 100644
--- a/common.gypi
+++ b/common.gypi
@@ -191,9 +191,9 @@
               'MSVC_runtimeType': 2   # MultiThreadedDLL (/MD)
             }],
             ['clang==1', {
-              'lto': ' -flto ', # Clang
+              'lto': ' -flto=thin ', # Clang
             }, {
-              'lto': ' -flto=4 -ffat-lto-objects ', # GCC
+              'lto': ' -flto=auto -fno-fat-lto-objects ', # GCC
             }],
           ],
         },
diff --git a/deps/cares/configure b/deps/cares/configure
index c2318a1c..a8ddbbcd 100755
--- a/deps/cares/configure
+++ b/deps/cares/configure
@@ -17166,7 +17166,6 @@ _LT_EOF
 
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
-*\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff --git a/deps/cares/m4/libtool.m4 b/deps/cares/m4/libtool.m4
index e7b68334..27a30926 100755
--- a/deps/cares/m4/libtool.m4
+++ b/deps/cares/m4/libtool.m4
@@ -7563,7 +7563,6 @@ _LT_EOF
 
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
-*\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff --git a/deps/openssl/openssl-cli.gypi b/deps/openssl/openssl-cli.gypi
index da03938f..00057f89 100644
--- a/deps/openssl/openssl-cli.gypi
+++ b/deps/openssl/openssl-cli.gypi
@@ -23,7 +23,7 @@
     }],
     # Avoid excessive LTO
     ['enable_lto=="true"', {
-      'ldflags': [ '-fno-lto' ],
+      'ldflags': [],
     }],
     ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
       'msvs_settings': {
diff --git a/deps/openssl/openssl.gyp b/deps/openssl/openssl.gyp
index 4e16412a..779d194b 100644
--- a/deps/openssl/openssl.gyp
+++ b/deps/openssl/openssl.gyp
@@ -84,7 +84,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/node.gyp b/node.gyp
index 8a6ab2cb..96be1797 100644
--- a/node.gyp
+++ b/node.gyp
@@ -1264,7 +1264,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
       ],
     }, # fuzz_env
@@ -1312,7 +1312,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
       ],
     }, # fuzz_ClientHelloParser.cc
@@ -1370,7 +1370,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
       ],
     }, # fuzz_strings
@@ -1482,7 +1482,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1555,7 +1555,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1642,7 +1642,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1778,7 +1778,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/tools/icu/icu-generic.gyp b/tools/icu/icu-generic.gyp
index f49b4ddb..40187c33 100644
--- a/tools/icu/icu-generic.gyp
+++ b/tools/icu/icu-generic.gyp
@@ -454,7 +454,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -481,7 +481,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -509,7 +509,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -536,7 +536,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/tools/v8_gypfiles/d8.gyp b/tools/v8_gypfiles/d8.gyp
index 5b47ebf1..a77f8de6 100644
--- a/tools/v8_gypfiles/d8.gyp
+++ b/tools/v8_gypfiles/d8.gyp
@@ -65,7 +65,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 7953fa44..a4e8d83c 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -1751,7 +1751,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1823,7 +1823,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1851,7 +1851,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1900,7 +1900,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1948,7 +1948,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1978,7 +1978,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -2057,7 +2057,7 @@
         ],
         'conditions': [
           ['enable_lto=="true"', {
-            'cflags_cc': [ '-fno-lto' ],
+            'cflags_cc': [],
           }],
           ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
             'msvs_settings': {
