diff --git a/common.gypi b/common.gypi
index dbd6bf9d..22d76117 100644
--- a/common.gypi
+++ b/common.gypi
@@ -189,9 +189,9 @@
               'MSVC_runtimeType': 2   # MultiThreadedDLL (/MD)
             }],
             ['clang==1', {
-              'lto': ' -flto ', # Clang
+              'lto': ' -flto=thin ', # Clang
             }, {
-              'lto': ' -flto=4 -ffat-lto-objects ', # GCC
+              'lto': ' -flto=auto -fno-fat-lto-objects ', # GCC
             }],
           ],
         },
diff --git a/deps/cares/configure b/deps/cares/configure
index f91e2c95..84a09a59 100755
--- a/deps/cares/configure
+++ b/deps/cares/configure
@@ -17166,7 +17166,6 @@ _LT_EOF
 
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
-*\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff --git a/deps/cares/m4/libtool.m4 b/deps/cares/m4/libtool.m4
index e7b68334..27a30926 100755
--- a/deps/cares/m4/libtool.m4
+++ b/deps/cares/m4/libtool.m4
@@ -7563,7 +7563,6 @@ _LT_EOF
 
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
-*\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff --git a/deps/openssl/openssl-cli.gypi b/deps/openssl/openssl-cli.gypi
index da03938f..00057f89 100644
--- a/deps/openssl/openssl-cli.gypi
+++ b/deps/openssl/openssl-cli.gypi
@@ -23,7 +23,7 @@
     }],
     # Avoid excessive LTO
     ['enable_lto=="true"', {
-      'ldflags': [ '-fno-lto' ],
+      'ldflags': [],
     }],
     ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
       'msvs_settings': {
diff --git a/deps/openssl/openssl.gyp b/deps/openssl/openssl.gyp
index 4d8d16f0..3a770c37 100644
--- a/deps/openssl/openssl.gyp
+++ b/deps/openssl/openssl.gyp
@@ -75,7 +75,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/node.gyp b/node.gyp
index b129c3db..9c685df4 100644
--- a/node.gyp
+++ b/node.gyp
@@ -1229,7 +1229,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
       ],
     }, # fuzz_env
@@ -1278,7 +1278,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
       ],
     }, # fuzz_ClientHelloParser.cc
@@ -1337,7 +1337,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
       ],
     }, # fuzz_strings
@@ -1449,7 +1449,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1523,7 +1523,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1610,7 +1610,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1746,7 +1746,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/tools/icu/icu-generic.gyp b/tools/icu/icu-generic.gyp
index f49b4ddb..40187c33 100644
--- a/tools/icu/icu-generic.gyp
+++ b/tools/icu/icu-generic.gyp
@@ -454,7 +454,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -481,7 +481,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -509,7 +509,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -536,7 +536,7 @@
       'conditions': [
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/tools/v8_gypfiles/d8.gyp b/tools/v8_gypfiles/d8.gyp
index 5b47ebf1..a77f8de6 100644
--- a/tools/v8_gypfiles/d8.gyp
+++ b/tools/v8_gypfiles/d8.gyp
@@ -65,7 +65,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 945aa0aa..38054a4e 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -1738,7 +1738,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1810,7 +1810,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1838,7 +1838,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1887,7 +1887,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1935,7 +1935,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -1965,7 +1965,7 @@
         }],
         # Avoid excessive LTO
         ['enable_lto=="true"', {
-          'ldflags': [ '-fno-lto' ],
+          'ldflags': [],
         }],
         ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
           'msvs_settings': {
@@ -2043,7 +2043,7 @@
         ],
         'conditions': [
           ['enable_lto=="true"', {
-            'cflags_cc': [ '-fno-lto' ],
+            'cflags_cc': [],
           }],
           ['node_with_ltcg=="true" or enable_lto=="true" or enable_thin_lto=="true"', {
             'msvs_settings': {
